### PR TITLE
daemon/snapshotter: Align mounter implementations

### PR DESCRIPTION
**Pull request by vvoland**
**- What I did**
Change the non-refcounted implementation to perform the mount using the same identity and access right. They should be the same regardless if we're refcounting or not.

This also allows to refactor `refCountMounter` into a mounter decorator.

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by rumpl — APPROVED** [review]

File: daemon/snapshotter/mount.go [hunk 1]
```diff
@@ -145,3 +138,7 @@ func (m mounter) Mount(mounts []mount.Mount, containerID string) (string, error)
 func (m mounter) Unmount(target string) error {
 	return unmount(target)
 }
+
+func (m mounter) target(containerID string) string {
+	return filepath.Join(m.home, mountsDir, m.snapshotter, containerID)
```
Comment (thaJeztah) [inline daemon/snapshotter/mount.go:143]: Looks like this is now the only location where `mountsDir` is used; we could either move the const close this, or just inline `"rootfs"`

**Review by thaJeztah — APPROVED** [review]
LGTM; left one suggestion; let me know if you want to update before merge

File: daemon/snapshotter/mount.go (hunk 1 above)
Comment (vvoland) [reply daemon/snapshotter/mount.go:143]: Can update quickly

Comment (vvoland) [reply daemon/snapshotter/mount.go]: Done.

Comment (thaJeztah) [reply daemon/snapshotter/mount.go]: saw it, thanks!

**Pull request merged**